### ISSUE TITLE
std::optional::value unavailable on earlier macOS versions

### ISSUE THREAD

**Issue by octobanana** [octobanana/peaclock#2]
@octobanana That problem is fixed, but the earlier MacOS versions face another problem:
`error: call to unavailable member function 'value': introduced in macOS 10.14`

A possible workaround is to remove the optional regex matches from `string.cc` and `string.hh`. Something like that:
```
-std::optional<std::vector<std::string>> match(std::string const& str, std::regex rx);
+std::vector<std::string> match(std::string const& str, std::regex rx);
```

The logic of the optional regex matches in `peacock.cc` should be refactored then, as there will be no more `match_opt`'s.

I have tried it locally just now, but the result seems a bit strange. There were no errors during the compilation and the clock runs just fine, the digital numbers are changing, but the binary ones are all enabled. Maybe something else went wrong, or my changes were incorrect.

_Originally posted by @Dluzhnevsky in https://github.com/octobanana/peaclock/issues/1#issuecomment-453676936_

**Comment by octobanana** [comment]
@Dluzhnevsky It sounds like the compiler being used doesn't fully support the C++17 standard. What compiler and version are you using?

According to [cppreference](https://en.cppreference.com/w/cpp/compiler_support), `std::optional` has support since gcc 7 and clang 4.

**Comment by Dluzhnevsky** [comment]
@octobanana Yeah, it is the C++17 support problem. I've heard a lot of people complaining about it.

I am using **clang: Apple LLVM version 10.0.0** (clang-1000.11.45.5). And it is still lacking the C++17 support.

**Comment by octobanana** [comment]
Unfortunately, supporting compilers that do not implement the C++17 standard is not a goal of this project.

Either updating to __macOS 10.14__ or installing a more recent compiler, such as __llvm__ or __gcc__, should fix the issue.

As installing a compiler is not within the scope of this project, this issue is being closed. Please feel free to reopen or create a new issue if you have any other questions.

**Comment by Czarek-Mojang** [comment]
I am using latest available clang `11.0.3` and running on Catalina `15.5` with all latest updates to comand line tools and I am getting the same error:

```
/third_party/stduuid/include/stduuid/uuid.h:900:47: error: 'value' is unavailable: introduced in
      macOS 10.14
            memcpy(&data[10], &device_address.value()[0], 6);
                                              ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/optional:952:27: note: 'value' has been explicitly marked
      unavailable here
    constexpr value_type& value() &
```

I am building with base SDK `10.15` on latest Catalina with newest clang, however I have set Target SDK to `10.10`, because my application needs to run on older OSes.

Should I create a new issue?
